### PR TITLE
Add Bitbucket access key module

### PR DESCRIPTION
**Pull request by zyv**
##### SUMMARY
Manages [Bitbucket](https://bitbucket.org) access keys.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
`bitbucket_access_key`, `bitbucket_deploy_key`

##### ADDITIONAL INFORMATION
This module is a re-implementation of the `bitbucket_deploy_key` module merged in #27467.

Our new implementation uses the recommended authentication way (OAuth) instead of requiring the user to specify his account password.

Also it uses modern API v2, while the old module is using old API v1, which is going to be **[removed  permanently on 29 April 2019](https://confluence.atlassian.com/bitbucket/repositories-endpoint-1-0-296092719.html)**, so the module will stop working.

### CODE REVIEW

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    type: str
+    required: true
+  username:
+    description:
+      - The repository owner.
+    type: str
+    required: true
+  key:
+    description:
+      - SSH public key.
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
      - The SSH public key.
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    type: str
+  label:
+    description:
+      - Key label.
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
      - The key label.
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    type: str
+    required: true
+  state:
+    description:
+      - Indicates desired state of the deploy key.
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
      - Indicates desired state of the access key.
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    type: str
+    required: true
+    choices: [ absent, present ]
+notes:
+  - Bitbucket OAuth consumer key and secret can be obtained from Bitbucket profile -> Settings -> Access Management -> OAuth.
+  - Bitbucket OAuth consumer should have permissions to read and admin the account repositories.
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
  - Bitbucket OAuth consumer should have permissions to read and administrate account repositories.
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  - Check mode is supported.
+'''
+
+EXAMPLES = r'''
+- name: Create deploy key
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
- name: Create access key
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+  bitbucket_access_key:
+    repository: 'bitbucket-repo'
+    username: bitbucket_username
+    key: '{{lookup("file", "bitbucket.pub") }}'
+    label: 'Bitbucket'
+    state: present
+
+- name: Delete deploy key
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
- name: Delete access key
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+'''
+
+RETURN = r''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.source_control.bitbucket import BitbucketHelper
+
+error_messages = {
+    'required_key': '`key` is required when the `state` is `present`',
+    'required_permission': 'OAuth consumer `client_id` should have permissions to read and admin the repository',
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
    'required_permission': 'OAuth consumer `client_id` should have permissions to read and administrate the repository',
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+
+DOCUMENTATION = r'''
+---
+module: bitbucket_access_key
+short_description: Manages Bitbucket repository deploy keys
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
short_description: Manages Bitbucket repository access keys
```

File: lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+description:
+  - Manages Bitbucket repository deploy keys.
```
Comment (zyv) [inline lib/ansible/modules/source_control/bitbucket/bitbucket_access_key.py]:
```suggestion
  - Manages Bitbucket repository access keys (also called deploy keys).
```

**Review by zyv — COMMENTED** [review]
Suggested wording changes.

**Review by catcombo — APPROVED** [review]

**Review by dagwieers — APPROVED** [review]
LGTM, and includes unit tests

**Pull request merged**